src/site/antora/modules/ROOT/pages/manual/garbagefree.adoc

```diff
@@ -14,565 +14,230 @@
+* xref:manual/filters.adoc#TimeFilter[`TimeFilter`] (garbage-free since `2.8` except when range must be recalculated once per day)
 
-|%m, %msg, %message
-|Log message (garbage-free unless message text
-contains '${')
+Any other filter not shared in the above list is not garbage-free.
 
-|%marker
-|The full name of the marker (including parents) - garbage-free
-since 2.8
+[#core-limitations]
+=== Limitations
 
-|%markerSimpleName
-|The simple name of the marker (not including
-parents)
+There are certain caveats associated with the configuration of garbage-free logging:
 
-|%maxLen, %maxLength
-|Truncates another pattern to some maximum number of
-characters - garbage-free since 2.8
+[#core-limitation-properties]
+`<Properties>` section::
 
-|%n
-|The platform-dependent line separator
+A configuration containing xref:manual/configuration.adoc#PropertySubstitution[a `<Properties>` section] will result in temporary objects being created during steady-state logging.
 
-|%N, %nano
-|System.nanoTime() when the event was logged
+[#core-limitation-async-logger-wait-strategy]
+Asynchronous logger wait strategies::
 
-|%notEmpty\{pattern}, %varsNotEmpty\{pattern},
-%variablesNotEmpty\{pattern}
-|Outputs the result of evaluating the
-pattern if and only if all variables in the pattern are not empty -
-garbage-free since 2.8
+As of version `2.18.0`, the default xref:manual/async.adoc[asynchronous logger] wait strategy (i.e., `Timeout`) is garbage-free while running against both LMAX Disruptor 3 and 4.
+See xref:manual/async.adoc#log4j2.asyncLoggerWaitStrategy[`log4j2.asyncLoggerWaitStrategy`] for details on predefined wait strategies.
 
-|%p, %level
-|The level of the logging event
+[#api]
+== Log4j API configuration and usage
 
-|%r, %relative
-|The number of milliseconds elapsed since the JVM was
-started until the creation of the logging event - garbage-free since 2.8
+In order to have a garbage-free Log4j API, you need to
 
-|%sn, %sequenceNumber
-|A sequence number that will be incremented in
-every event - garbage-free since 2.8
+* xref:#api-config[configure it using properties],
+* and xref:#api-usage[use garbage-free API methods].
 
-|%style\{pattern}{ANSI style}
-|Style the message - garbage-free since
-2.7 (unless the nested pattern is not garbage-free)
+[#api-config]
+=== Configuration
 
-|%T, %tid, %threadId
-|The ID of the thread that generated the logging
-event
+Garbage-free logging can be configured for Log4j API using properties listed below.
+(See xref:manual/configuration.adoc[] on details how you can set these properties.)
 
-|%t, %tn, %thread, %threadName
-|The name of the thread that generated
-the logging event
+include::partial$manual/systemproperties/properties-garbage-collection.adoc[leveloffset=+2,tag=api]
 
-|%tp
-|The priority of the thread that generated the logging event
+include::partial$manual/systemproperties/properties-thread-context.adoc[leveloffset=+2,tag=gcfree]
```

Comment:
As in the remark above: the algorithm that interprets which `ThreadContextMap` implementation to use has been moved from the `ThreadContextMapFactory` class (in `log4j-api`) to implementations of `Provider` (in `log4j-core` and the other Log4j API implementations).

That is why I would consider this part of Log4j Core configuration.

Reply:
Moved all shared properties to the `Log4j Core configuration > Properties` section in ea573b384d800a076b45e5c7e21d6b00eadc59fc.